archive-cleaner

```diff
@@ -32,10 +31,11 @@ def get_sourceball(path):
     date = m.group(0)
 
     pkgname = basename(path)
+    source_ext = '.src.{}'.format('.'.join(pkgname.split('.')[1:]))
```

Comment:
This approach doesn't work, packages can have arbitrary amount of dots in their name

Reply:
If path is a path doesn't basename give packagename.tar,gz or whatever? That's what I thought

Reply:
Yeah my reply was stupid, I get what you mean now

Reply:
```
def get_sourceball(path):
    print(path)
    m = search('\d{4}/\d{2}/\d{2}', path)
    date = m.group(0)

    pkgname = basename(path)
    pkg_ext = next(filter(lambda ext: ext in pkgname, PKG_EXTENSIONS), None)
    if pkg_ext is None:
        # should not happen
        return None
    source_ext = f'.src.{pkg_ext}'
    pkgver, pkgrel, _ = pkgname.split('-')[-3:]
    pkgname, _ = pkgname.split(f'-{pkgver}-{pkgrel}')

    return f'/srv/archive/repos/{date}/sources/packages/{pkgname}-{pkgver}-{pkgrel}{source_ext}'
```
I'm not sure what do do in the case path has not valid extension, I mean that shouldn't really happen, I guess I might return a path with .src.tar.gz extension that later is found to be non-existent so I don't have to touch all the other code and check for not None

Reply:
that shouldn't really happen, so in case, we could just abort and throw an exception i guess?
you can simplify the code like
```python
pkg_ext = list(filter(lambda ext: ext in pkgname, PKG_EXTENSIONS))
if not pkg_ext:
    raise ...
```

Reply:
I thought using iterators next would be more efficient than collecting into a list, but sure.
I did `raise ValueError(f'Expecting path to end with one of these extensions {" ".join(PKG_EXTENSIONS)}. {path} given')`

Reply:
I'll go with the list just to keep consistent as in other places I used list actually